R/equivalence_test.R

```diff
@@ -48,21 +53,42 @@ equivalence_test.numeric <- function(posterior, bounds = "default", ci = .95, ve
     )
   }
 
+  out$HDI_low <- attr(rope_data, "HDI_area", exact = TRUE)[1]
+  out$HDI_high <- attr(rope_data, "HDI_area", exact = TRUE)[2]
+
+  class(out) <- c("equivalence_test", class(out))
   out
 }
 
 
+#' @export
+print.equivalence_test <- function(x, ...) {
+  cat("# Test for Practical Equivalence\n\n")
+  cat(sprintf("  ROPE: [%.2f %.2f]\n\n", x$ROPE_low[1], x$ROPE_high[1]))
+  ci <- x$CI[1]
+
+  x$ROPE_Percentage <- sprintf("%.2f%%", x$ROPE_Percentage)
+  x$HDI <- sprintf("[%.2f %.2f]", x$HDI_low, x$HDI_high)
+
+  x <- x[, intersect(c("Parameter", "ROPE_Equivalence", "ROPE_Percentage", "HDI"), colnames(x))]
+
+  colnames(x)[which(colnames(x) == "ROPE_Equivalence")] <- "H0"
+  colnames(x)[which(colnames(x) == "ROPE_Percentage")] <- "Inside ROPE"
+  colnames(x)[ncol(x)] <- sprintf("%i%% HDI", ci)
```

Comment:
This doesn't work in the case of multiple CIs:
```r
equivalence_test(posterior = rnorm(1000, 1, 1), ci = c(.50, .99))
```
```
# Test for Practical Equivalence

  ROPE: [-0.10 0.10]

        H0 Inside ROPE     50% HDI
  rejected       0.00% [0.25 1.58]
 undecided       4.24% [0.25 1.58]
```
One workaround would be moving the desired CI proportion in a separate column, this way they could differ from row to row.